### PR TITLE
 [#213]: tests methods of MessageTypeDetails

### PR DESCRIPTION
**Pull request by xieyuschen**
## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked in the [References](#references) section
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

Updates #213

### CODE REVIEW

File: iceoryx2/src/service/static_config/message_type_details.rs [hunk 1]
```diff
@@ -266,4 +266,130 @@ mod tests {
...
+        let expected = 36;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 48;
+        let sut = details.payload_layout(3);
+        assert_that!(sut.size(), eq expected);
+    }
+
+    #[test]
+    // test_sample_layout tests the sample layout for combinations of different types.
+    fn test_sample_layout() {
```
Comment (xieyuschen) [inline iceoryx2/src/service/static_config/message_type_details.rs:303]: I haven't found a better way to test it, so I calculated them in mind and test the layout for different architecture. Do you think it's fine or better idea?

**Comment by xieyuschen** [comment]
@elfenpiff @elBoberido i don't have the permission to assign the reviewer:( so at you directly

File: iceoryx2/src/service/static_config/message_type_details.rs [hunk 2]
```diff
@@ -266,4 +266,130 @@ mod tests {
...
+                alignment: 2 * ALIGNMENT,
+            },
+            payload: TypeDetail {
+                variant: TypeVariant::FixedSize,
+                type_name: "i64".to_string(),
+                size: 8,
+                alignment: 2 * ALIGNMENT,
+            },
+        };
+        // smaller to bigger is allowed.
```
Comment (elfenpiff) [inline iceoryx2/src/service/static_config/message_type_details.rs:387]: Could you create separate tests out of it with an expressive name like your comment here. We try that every test verifies just one aspect in isolation - when it makes sense.

**Review by elfenpiff — COMMENTED** [review]
Just split the one test case up into several ones and we are good to merge!

File: iceoryx2/src/service/static_config/message_type_details.rs (hunk 1 above)
Comment (elfenpiff) [reply iceoryx2/src/service/static_config/message_type_details.rs:303]: This is fine. Sometimes we have to go this route...

File: iceoryx2/src/service/static_config/message_type_details.rs (hunk 2 above)
Comment (xieyuschen) [reply iceoryx2/src/service/static_config/message_type_details.rs:392]: done in fa7e803f96655d146988a67f0a70215ce0f95880

**Comment by xieyuschen** [comment]
> Just split the one test case up into several ones and we are good to merge!

cheers! @elfenpiff

**Comment by orecham** [comment]
@xieyuschen Could you add `[#203]` to the commit message `split the big test_is_compatible_to into small tests`

Not sure why the preflight-check didn't catch this 🤔

**Comment by xieyuschen** [comment]
> @xieyuschen Could you add `[#213]` to the commit message `split the big test_is_compatible_to into small tests`
> 
> Not sure why the preflight-check didn't catch this 🤔

done, thanks for reminding.

**Comment by xieyuschen** [comment]
@elfenpiff @orecham looks like the macos pipeline is a bit unstable and it failed to setup rust. The other pipelines passed.

**Review by elfenpiff — APPROVED** [review]
@xieyuschen again, thanks for the work you have done here.

**Comment by elfenpiff** [comment]
@xieyuschen The next iceoryx community meetup is online on the 5th of November: https://github.com/eclipse-iceoryx/iceoryx/wiki/2024-11-05-Eclipse-iceoryx-developer-meetup

It would be nice to get to know you there. I know it is still a while, but you already committed some great contributions so it would be a pleasure to get to know you in person if you like. Also we could discuss if you would be interested in driving a feature from our roadmap - something  that might interest you or you always wanted to try.

Great to have you on board!

**Comment by xieyuschen** [comment]
> @xieyuschen The next iceoryx community meetup is online on the 5th of November: https://github.com/eclipse-iceoryx/iceoryx/wiki/2024-11-05-Eclipse-iceoryx-developer-meetup
> 
> It would be nice to get to know you there. I know it is still a while, but you already committed some great contributions so it would be a pleasure to get to know you in person if you like. Also we could discuss if you would be interested in driving a feature from our roadmap - something that might interest you or you always wanted to try.
> 
> Great to have you on board!

cool, I'm looking forward to meeting you guys in the meeting and learning more about future plans.

**Pull request merged**